### PR TITLE
[Prometheus] Remove shared project and move the shared code under listener project

### PR DESCRIPTION
**Pull request by Yun-Ting**
Related to:

https://github.com/open-telemetry/opentelemetry-dotnet/pull/3430#discussion_r930521084
https://github.com/open-telemetry/opentelemetry-dotnet/pull/3497#discussion_r931572797
https://github.com/open-telemetry/opentelemetry-dotnet/pull/3430#discussion_r932669508

## Changes
Trying out another project setup for prometheus.

Move the files that were previously in `OpenTelemetry.Exporter.Prometheus.Shared` to be under the `OpenTelemetry.Exporter.Prometheus.HttpListener` project. 
And removed `OpenTelemetry.Exporter.Prometheus.Shared` project and its corresponding test project.

`OpenTelemetry.Exporter.Prometheus.AspNetCore` now reference the shared code from the listener project.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/OpenTelemetry.Exporter.Prometheus.AspNetCore.csproj [hunk 1]
```diff
@@ -19,12 +19,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Prometheus.Shared\PrometheusCollectionManager.cs" Link="Includes/PrometheusCollectionManager.cs" />
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Prometheus.Shared\PrometheusExporter.cs" Link="Includes/PrometheusExporter.cs" />
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Prometheus.Shared\PrometheusExporterEventSource.cs" Link="Includes/PrometheusExporterEventSource.cs" />
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Prometheus.Shared\PrometheusExporterOptions.cs" Link="Includes/PrometheusExporterOptions.cs" />
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Prometheus.Shared\PrometheusSerializer.cs" Link="Includes/PrometheusSerializer.cs" />
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Prometheus.Shared\PrometheusSerializerExt.cs" Link="Includes/PrometheusSerializerExt.cs" />
+    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Prometheus.HttpListener\PrometheusCollectionManager.cs" Link="Includes/PrometheusCollectionManager.cs" />
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Prometheus.AspNetCore/OpenTelemetry.Exporter.Prometheus.AspNetCore.csproj:22]: Maybe put these files under a folder named `Shared` or `Internal`?

**Comment by reyang** [comment]
Are the shared code actually removed from `OpenTelemetry.Exporter.Prometheus.Shared` by this PR?

**Review by reyang — APPROVED** [review]
Approving given it is better than the current layout. I think the public class under shared folder is something that should be fixed in the next PR.

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/OpenTelemetry.Exporter.Prometheus.AspNetCore.csproj (hunk 1 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Exporter.Prometheus.AspNetCore/OpenTelemetry.Exporter.Prometheus.AspNetCore.csproj]: I've moved the shared files between AspNetCore and the HttpListener projects that reside in the `OpenTelemetry.Exporter.Prometheus.HttpListener` project to a folder called `Shared`. And from the AspNetCore project, grouped the referenced shared files in the `Include` folder.

**Comment by Yun-Ting** [comment]
> Are the shared code actually removed from `OpenTelemetry.Exporter.Prometheus.Shared` by this PR?

It is now :-)

**Pull request merged**